IHP/View/CSSFramework.hs

```diff
@@ -93,52 +93,104 @@ instance Default CSSFramework where
+                                                disabled={disabled}
+                                                autofocus={autofocus}
+                                            />
+
+                                            {validationResult}
+                                            {helpText}
+                                    |]
+
 
                     renderSelectField :: FormField -> Blaze.Html
                     renderSelectField formField@(FormField {fieldType, fieldName, placeholder, fieldLabel, fieldValue, fieldInputId, validatorResult, fieldClass, disabled, disableLabel, disableValidationResult, fieldInput, labelClass, required, autofocus }) =
-                        formGroup do
-                            unless disableLabel [hsx|<label class={labelClass} for={fieldInputId}>{fieldLabel}</label>|]
-                            H.select ! A.name fieldName ! A.id (cs fieldInputId) ! A.class_ (cs $ classes [inputClass, (inputInvalidClass, isJust validatorResult), (fieldClass, not (null fieldClass))]) ! A.value (cs fieldValue) !? (disabled, A.disabled "disabled") !? (required, A.required "required") !? (autofocus, A.autofocus "autofocus") $ do
-                                let isValueSelected = isJust $ find (\(optionLabel, optionValue) -> optionValue == fieldValue) (options fieldType)
-                                (if isValueSelected then Blaze.option else Blaze.option ! A.selected "selected")  ! A.disabled "disabled" $ Blaze.text placeholder
-                                forEach (options fieldType) $ \(optionLabel, optionValue) -> (let option = Blaze.option ! A.value (cs optionValue) in (if optionValue == fieldValue then option ! A.selected "selected" else option) $ cs optionLabel)
-                            validationResult
-                            helpText
+                        formGroup element
+                        where
+                            label = unless disableLabel [hsx|<label class={labelClass} for={fieldInputId}>{fieldLabel}</label>|]
+
+                            element = [hsx|
+                                        {label}
+                                        <select
+                                            name={fieldName}
+                                            id={fieldInputId}
+                                            class={classes [inputClass, (inputInvalidClass, isJust validatorResult), (fieldClass, not (null fieldClass))]}
+                                            value={fieldValue}
+                                            disabled={disabled}
+                                            required={required}
+                                            autofocus={autofocus}
+                                        >
+                                            <option selected={not isValueSelected} disabled={True}>{placeholder}</option>
+                                            {forEach (options fieldType) (getOption)}
```

Comment:
There are now spaces that appear on the test, but I don't know where they are coming from.

![image](https://user-images.githubusercontent.com/125707/142739093-fff519c0-b6ab-40c3-bc5c-e7588252ff26.png)

Reply:
These whitespaces come from HSX expression like `<some-tag> {someVariableExpr}`. In this case HSX cannot safely remove the full whitespace between these two nodes as  we don't know whats inside `someVariableExpr` at compile-time.

You could rewrite the code like this:

```haskell
<option selected={not isValueSelected} disabled={True}>{placeholder}</option>{forEach (options fieldType) (getOption)}
```

Doesn't look nice, but get rids of the space there then :)

Reply:
Those spaces are actually between the closing `</select>` and the closing `</div>`. They are the empty spaces as a result of 

```
{validationResult}
{helpText}
```

I could solve it by adding two spaces to the expected result, but not sure it's the right way.

Reply:
Try to collapse this to `{validationResult}{helpText}`. Does that remove the spaces?

Reply:
Removes one space, but one more to go :)

![image](https://user-images.githubusercontent.com/125707/142771102-f66668d7-e652-4a37-b162-1cd04ba62da1.png)

Reply:
@mpscholten should I add spaces to the expected result, or do you have a better idea?

Reply:
In that case let's just add the space to the test case 👍